### PR TITLE
#162: Added negation for `haveRawType`

### PR DESCRIPTION
**Pull request by asbachb**

### CODE REVIEW

File: archunit/src/test/java/com/tngtech/archunit/lang/syntax/elements/FieldsShouldTest.java [hunk 1]
```diff
@@ -60,6 +60,24 @@ public void property_predicates(ArchRule rule) {
         assertThat(actualFields).containsOnly(FIELD_B, FIELD_C, FIELD_D);
     }
 
+    @DataProvider
+    public static Object[][] restricted_property_rule_ends2() {
```
Comment (asbachb) [inline archunit/src/test/java/com/tngtech/archunit/lang/syntax/elements/FieldsShouldTest.java]: I know the name is not ideal. Maybe you could propose a better name.

File: archunit/src/test/java/com/tngtech/archunit/lang/syntax/elements/FieldsShouldTest.java [hunk 2]
```diff
@@ -60,6 +60,24 @@ public void property_predicates(ArchRule rule) {
...
+        return testForEach(
+                fields().should().notHaveRawType(String.class),
+                fields().should().notHaveRawType(String.class.getName()),
+                fields().should().notHaveRawType(equivalentTo(String.class).as(String.class.getName())));
+    }
+
+    @Test
+    @UseDataProvider("restricted_property_rule_ends2")
+    public void property_predicates2(ArchRule rule) {
```
Comment (asbachb) [inline archunit/src/test/java/com/tngtech/archunit/lang/syntax/elements/FieldsShouldTest.java]: I know the name is not ideal. Maybe you could propose a better one.

File: archunit/src/main/java/com/tngtech/archunit/lang/syntax/elements/FieldsShould.java [hunk 3]
```diff
@@ -44,6 +44,26 @@
     @PublicAPI(usage = ACCESS)
     CONJUNCTION haveRawType(Class<?> type);
 
+    /**
+     * Asserts that fields not have a certain raw type.
```
Comment (hankem) [inline archunit/src/main/java/com/tngtech/archunit/lang/syntax/elements/FieldsShould.java]: Maybe _"Asserts that fields **do** not have a certain raw type."_ ?
(Same below.)

Comment (asbachb) [reply archunit/src/main/java/com/tngtech/archunit/lang/syntax/elements/FieldsShould.java]: Good idea!

**Comment by codecholeric** [comment]
Hey, thanks for the initiative :smiley: 
Can you do me one favor and sign-off your commit? See https://github.com/TNG/ArchUnit/blob/master/CONTRIBUTING.md
I think the easiest is to reword the commit message of your old commit and just add the signed-off line (same line you would get with `git commit -s`).
Thanks!!

**Comment by asbachb** [comment]
> Hey, thanks for the initiative smiley
> Can you do me one favor and sign-off your commit? See https://github.com/TNG/ArchUnit/blob/master/CONTRIBUTING.md
> I think the easiest is to reword the commit message of your old commit and just add the signed-off line (same line you would get with `git commit -s`).
> Thanks!!

Sorry I missed that. I'll have a more detailed look into it later on.

**Comment by asbachb** [comment]
> Hey, thanks for the initiative smiley
> Can you do me one favor and sign-off your commit? See https://github.com/TNG/ArchUnit/blob/master/CONTRIBUTING.md
> I think the easiest is to reword the commit message of your old commit and just add the signed-off line (same line you would get with `git commit -s`).
> Thanks!!

@codecholeric I added the sign-off line, but I'm quire unsure if it needs to match with the mail address used by the commit (I normally use github pseudo mail address) or is a real mail address needed?

**Comment by asbachb** [comment]
Retry Deepcode

**Comment by codecholeric** [comment]
@asbachb That should be fine, in the end this is just for you to verify that you have the rights to make this contribution (e.g. because it's your own code on your own time) and that you put it under the project's license.
I would assume that we could reach you through the contact details at https://www.impl.it/ if need should ever arise :wink:

**Comment by codecholeric** [comment]
Done :smiley: Thanks again for your contribution!!

**Pull request merged**